daemon/src/surface.rs

```diff
@@ -207,7 +209,45 @@ impl Surface {
                     bail!("Directory {path:?} is empty");
                 }
 
-                let img_path = files[rand::random::<usize>() % files.len()].clone();
+                // Set index for the various sorting methods
+                let index = match self.wallpaper_info.sorting {
+                    Sorting::Random => {
+                        rand::random::<usize>() % files.len()
+                    },
+                    Sorting::Ascending => {
+                        let idx = match files.binary_search(&self.current_img) {
+                            Ok(n) => n,
+                            Err(err) => {
+                                info!("Current image not found, defaulting to first image ({:?})", err);
+                                // set idx to > slice length so the guard sets it correctly for us
+                                files.len()
+                            }
+                        };
+                        if idx >= files.len() {
+                            0
+                        } else {
+                            idx + 1
```

Comment:
We don't know if `idx + 1` is still inside the vector range. Since we call:

```rust
let img_path = files.into_iter().nth(index).unwrap();
```

wpaperd might panic in that case.

Reply:
Also, we can use either `[]` operator or `get` to files without calling `into_iter`.

Reply:
One idea is to increase the index regardless and then check afterwards.